src/Domain/PasswordResetToken.php

```diff
@@ -43,14 +43,24 @@ class PasswordResetToken
      */
     private ?\DateTimeImmutable $used = null;
 
-    public function __construct(Person $person)
+    private function __construct(Person $person, Uuid $uuid)
     {
         $this->person = $person;
-        $this->secret = Uuid::v4();
+        $this->secret = $uuid;
         $this->created_at = new \DateTime();
         $this->updated_at = new \DateTime();
     }
 
+    public static function fromBase58(Person $person, string $base58Token): self
+    {
+        return new self($person, Uuid::fromBase58($base58Token));
+    }
+
+    public static function random(Person $person): self
+    {
+        return new self($person, Uuid::v4());
+    }
+
```

Comment:
See https://verraes.net/2014/06/named-constructors-in-php/

Reply:
@bdsl Why do we need two different static methods that do roughly the same thing? Why can't we simply choose to always return `new self($person, Uuid::fromBase58($base58Token));` for example?

Reply:
They don't really do the same thing.

One makes a random token which is what we need in prod, one makes a token deterministically which is much easier to use in unit tests. It might be nice to take the randomness (as well as use of other system resources like the clock) out of the domain layer, then we'd have to either put it in the Action or make somewhere else to put it.

If we just had a constructor doing `new self($person, Uuid::fromBase58($base58Token));` how would we use it in `CreatePasswordResetToken::action` ?